okio/src/jvmMain/kotlin/okio/AsyncTimeout.kt

```diff
@@ -250,77 +288,67 @@ open class AsyncTimeout : Timeout() {
      */
     private var head: AsyncTimeout? = null
 
-    private fun scheduleTimeout(node: AsyncTimeout, timeoutNanos: Long, hasDeadline: Boolean) {
-      AsyncTimeout.lock.withLock {
-        check(!node.inQueue) { "Unbalanced enter/exit" }
-        node.inQueue = true
-
-        // Start the watchdog thread and create the head node when the first timeout is scheduled.
-        if (head == null) {
-          head = AsyncTimeout()
-          Watchdog().start()
-        }
+    private fun insertIntoQueue(node: AsyncTimeout, timeoutNanos: Long, hasDeadline: Boolean) {
```

Comment:
Most of the changes in this function are just whitespace. I moved `withLock {}` to the caller.

<img width="223" alt="Screenshot 2023-12-16 at 10 03 42 AM" src="https://github.com/square/okio/assets/133019/7499b016-4f0f-450b-8b24-f7c301e608d5">